### PR TITLE
Data attribute in payloads

### PR DESCRIPTION
**Pull request by leplatrem**
**Do not merge before** #22

### CODE REVIEW

File: test/api_test.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ describe("Api", () => {
     describe("full URL", () => {
       var endpoints;
 
-      beforeEach(() => endpoints = api.endpoints({fullUrl: true}))
+      beforeEach(() => endpoints = api.endpoints({ fullUrl: true }))
```
Comment (Natim) [inline test/api_test.js]: We should choose if we want spaces between curly-brackets and property name and keep it to

Comment (leplatrem) [inline test/api_test.js]: Yes, I could not resist to change the few ones I saw... But there may be a way to configure jslint

Comment (Natim) [inline test/api_test.js]: I prefer ``{data: "toto"}`` rather than ``{ data: "toto" }`` but it is a matter of taste I guess.

@n1k0 what do you think?

Comment (n1k0) [inline test/api_test.js]: I don't have a strong opinion on this, really. Your call.

File: test/api_test.js [hunk 2]
```diff
@@ -124,20 +124,19 @@ describe("Api", () => {
...
-    it("should attach an If-Modified-Since header if lastModified is provided", () =>{
+    it("should attach an If-None-Match header if lastModified is provided", () =>{
       sandbox.stub(root, "fetch").returns(Promise.resolve());
-
       api.fetchChangesSince("articles", 42);
 
       sinon.assert.calledOnce(fetch);
       sinon.assert.calledWithMatch(fetch, /\?_since=42/, {
-        headers: {"If-Modified-Since": "42"}
+        headers: { "If-None-Match": '"42"' }
```
Comment (n1k0) [inline test/api_test.js]: Maybe reuse `quote` for this assertion.

File: test/api_test.js [hunk 3]
```diff
@@ -124,20 +124,19 @@ describe("Api", () => {
...
       });
     });
 
     it("should resolve with a result object", () => {
       sandbox.stub(root, "fetch").returns(
-        fakeServerResponse(200, {items: []}, {"Last-Modified": 41}));
+        fakeServerResponse(200, { data: [] }, { "ETag": '"41"' }));
```
Comment (n1k0) [inline test/api_test.js]: Ditto.

File: test/api_test.js [hunk 4]
```diff
@@ -217,36 +216,43 @@ describe("Api", () => {
...
         it("should map create & update requests", () => {
           expect(requestBody.requests[0]).eql({
-            "body": {
-              "id": 1,
-              "title": "foo",
+            body: {
+              data: { id: 1, title: "foo" },
             },
-            headers: {"If-Unmodified-Since": "42"},
+            headers: { "If-Match": '"42"' },
```
Comment (n1k0) [inline test/api_test.js]: Ditto.

File: test/api_test.js [hunk 5]
```diff
@@ -217,36 +216,43 @@ describe("Api", () => {
...
             path: "/v0/collections/articles/records/3",
           });
         });
 
-        it("should map batch delete requests for synced records", () => {
+        it("should map batch update requests for synced records", () => {
           expect(requestBody.requests[0]).eql({
             path: "/v0/collections/articles/records/1",
             method: "PUT",
+            headers: { "If-Match": '"42"' },
```
Comment (n1k0) [inline test/api_test.js]: Ditto.

File: test/api_test.js [hunk 6]
```diff
@@ -261,8 +267,8 @@ describe("Api", () => {
           requests = JSON.parse(fetch.getCall(0).args[1].body).requests;
         });
 
-        it("should send If-Unmodified-Since headers", () => {
-          expect(requests[0].headers).eql({"If-Unmodified-Since": "42"});
+        it("should send If-Match headers", () => {
+          expect(requests[0].headers).eql({"If-Match": '"42"'});
```
Comment (n1k0) [inline test/api_test.js]: Ditto.

**Pull request closed without merging**